consensus/istanbul/backend/engine.go

```diff
@@ -560,6 +561,41 @@ func (sb *Backend) SetChain(chain consensus.ChainReader, currentBlock func() *ty
 	sb.chain = chain
 	sb.currentBlock = currentBlock
 	sb.stateAt = stateAt
+
+	if bc, ok := chain.(*ethCore.BlockChain); ok {
+		// Start listeners
+		chainHeadCh := make(chan ethCore.ChainHeadEvent, 10)
```

Comment:
We can encapsulate both goroutines (maybe even have just one) behind a function named so that it conveys the nature of both routines... in fact, having that we don't need to have a `UpdateReplicaState` method, since it would an internal call here. and maybe the routine is `replicaStatusListener()` or something..

same with `newChainHead` do we need a method for that? i believe it was created just because it was a public interface.

Reply:
Updated. See `consensus/istanbul/backend/engine.go:572` I changed `NewChainHead` -> `newChainHead` because I assumed we added it. It could be reverted back to being a public interface if it's supposed to be part of that interface.